gdal/ogr/ogr_api.cpp

```diff
@@ -32,14 +32,53 @@
 #include "ogr_api.h"
 
 #include <cstddef>
+#include <regex>
 
 #include "cpl_error.h"
 #include "ogr_geometry.h"
+#include "ogr_geos.h"
 
 CPL_CVSID("$Id$")
 
 static bool bNonLinearGeometriesEnabled = true;
 
+/************************************************************************/
+/*                         OGRGetGEOSVersion()                          */
+/************************************************************************/
+
+/** \brief Get the GEOS version
+ *
+ * @param pnMajor Pointer to major version number, or NULL
+ * @param pnMinor Pointer to minor version number, or NULL
+ * @param pnPatch Pointer to patch version number, or NULL
+ * @since GDAL 3.4.0
+ */
+#ifdef HAVE_GEOS
+void OGRGetGEOSVersion(int *pnMajor, int *pnMinor, int *pnPatch) {
+    std::string verstring = GEOSversion();
+    auto tokenizer = std::regex("\\.");
```

Comment:
can use use instead to avoid std::regex ?
``const CPLStringList aosTokens(CSLTokenizeString2(verstring.c_str(), ".", 0))``